### ISSUE TITLE
Pushing the tenantId filter to subdocument model for populate

### ISSUE THREAD

**Issue by pavankumarb** [craftup/node-mongo-tenant#16]
Hi,
Thanks for the useful utility. 
Is there a way to push the tenantId filter down into Subdocument populate.
I tried to add _conditions in pre hook of the 'find' method but the object has no clue at this stage about its parent.  Find pre hook is called for each subdocument populate construct.

this.schema.pre('find', function(next) { 
//add here
});  
Another approach I tried, which seems to work but looks a bit cumbersome is in mongoose/lib/model.js code. Here I can fetch the parent object's tenantId and pass it to match clause of the subdocument.

Any recommendation on the right approach to implement this.

Best regards
Pav

**Comment by alrik** [comment]
@pavankumarb thanks for using mongo-tenant and for your feature request. I'm going to figure out a way in order to implement tenantId population into subdocuments. Expact a result within the next 2 weeks. But feel free to submit a PR in the meantime :)

**Comment by pavankumarb** [comment]
Thank you @alrik for a quick response. Appreciate it.

**Comment by alrik** [comment]
@pavankumarb Sorry, i didn't find the time to dig into this, yet :(

**Comment by pavankumarb** [comment]
No worries Alrik.

**Comment by osvathrobi** [comment]
Just putting it out there: i'd also need this. Thanks for making this package.

**Comment by alrik** [comment]
I'm on it. But did not find the time for digging in it, yet.

**Comment by baranga** [comment]
Hi, just tried to reproduce the issue. Added a unit test in PR #36 to cover the problem, at least how I understood it.

**Comment by baranga** [comment]
Possible workaround could look like this:
```js
ParentModel.byTenant('tenant1')
  .find()
  .populate({path: 'childs', model: ChildModel.byTenant('tenant1')})
  .exec(...);
```

**Comment by baranga** [comment]
@pavankumarb @osvathrobi PR #36 contains now a fix for the issue. Can you evaluate if that fixes your issues?

**Comment by baranga** [comment]
@pavankumarb @osvathrobi PR is merged to develop, release should be coming soon.

**Comment by alrik** [comment]
Released with version `1.6.0`
